pennylane/ops/op_math/linear_combination.py

```diff
@@ -100,21 +102,19 @@ class LinearCombination(Sum):
 
     def _flatten(self):
         # note that we are unable to restore grouping type or method without creating new properties
-        return (self._coeffs, self._ops, self.data), (self.grouping_indices,)
+        return self.terms(), (self.grouping_indices,)
 
     @classmethod
     def _unflatten(cls, data, metadata):
-        new_op = cls(data[0], data[1])
-        new_op._grouping_indices = metadata[0]  # pylint: disable=protected-access
-        new_op.data = data[2]
-        return new_op
+        return cls(data[0], data[1], grouping_indices=metadata[0])
 
     def __init__(
         self,
         coeffs,
         observables: List[Operator],
         simplify=False,
         grouping_type=None,
+        grouping_indices=None,
         method="rlf",
         _pauli_rep=None,
```

Comment:
This argument "looks private" in a public function, that's a bit funny :thinking:
Also, does not seem to be documented?

Reply:
Not documented as it's a "private" keyword argument.  When converting back from a pauli rep to an operator, we don't have to re-calculate the pauli rep, because that can get quite expensive. So not something a user should care about, but we need internally.  


Potentially we could make `grouping_indices` a private kwarg too.